src/java.base/share/classes/jdk/internal/util/ByteArray.java

```diff
@@ -41,12 +40,13 @@ public final class ByteArray {
     private ByteArray() {
     }
 
-    private static final VarHandle SHORT = create(short[].class);
-    private static final VarHandle CHAR = create(char[].class);
-    private static final VarHandle INT = create(int[].class);
-    private static final VarHandle FLOAT = create(float[].class);
-    private static final VarHandle LONG = create(long[].class);
-    private static final VarHandle DOUBLE = create(double[].class);
+    static final Unsafe UNSAFE = Unsafe.getUnsafe();
```

Comment:
I recommend documenting the reason to use Unsafe (early startup) and comment that regular users should use VarHandle instead.